lib/pipedrive/helpers/pagination.ex

```diff
@@ -45,16 +45,24 @@ defmodule Pipedrive.Helpers.Pagination do
     end
   end
 
-  defp put_new_url_param(opts, param, value) do
-    if get_in(opts, [:url_params, param]) do
-      opts
+  defp put_new_url_param(api_call_args, param, value) do
+    if get_in(get_opts(api_call_args), [:url_params, param]) do
+      api_call_args
     else
-      put_url_param(opts, [:url_params, param], value)
+      put_url_param(api_call_args, [:url_params, param], value)
     end
   end
 
-  def put_url_param(opts, param, value) do
-    put_in(opts, [:url_params, param], value)
+  defp put_url_param(api_call_args, param, value) do
+    update_opts(api_call_args, &put_in(&1, [:url_params, param], value))
+  end
+
+  defp get_opts(api_call_args) do
```

Comment:
@angelikatyborska Calling `Pagination.fetch_all/1` would now raise an error since in line 26 `Keyword.get`-is operating on `nil` if `api_call_args` defaults to `[]`, would it?

I'd propose to default to an empty list here:

```elixir
defp get_opts(api_call_args) do
  List.last(api_call_args) || []
end
```

Reply:
Hmm.. or maybe I just changed my mind and it would be better to default in `fetch_all` to a List with an empty (Keyword) list.

```elixir
def fetch_all(api_call, api_call_args \\ [Keyword.new()]) do
  …
end
```

Reply:
Good point. Yes, that would raise an error.

I changed the default arg to `fetch_all` to `[[]]`.